From 02103b5aefbb4f90c9b019f2c6b962f9b8613f55 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Thu, 24 Dec 2015 13:22:05 +0530
Subject: [PATCH 07/14] USBC-PHY: Introduce vbus notifier to avoid polling.

This patch will add a notifier for vbus change so that
the policy engine need not to poll for it when required.

Change-Id: Ibda83ff1abc895f4a67990a11478c86f58556310
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15496
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/451419
---
 drivers/usb/typec/phy-fusb300.c | 5 +++++
 include/linux/usb_typec_phy.h   | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index f5d7592..c05b9df 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -866,12 +866,17 @@ static void fusb300_handle_vbus_int(struct fusb300_chip *chip, int vbus_on)
 		}
 		complete(&chip->vbus_complete);
 
+		/* Nofity Detect driver */
 		atomic_notifier_call_chain(&phy->notifier,
 				 TYPEC_EVENT_VBUS, phy);
 		/* TOG_DONE will be used with FUSB302 */
+		/* Notify DPM */
+		dpm_handle_phy_event(phy, PHY_DPM_EVENT_VBUS_ON);
 	} else {
 		if (state == TYPEC_STATE_ATTACHED_UFP)
 			schedule_delayed_work(&chip->dfp_disconn_work, 0);
+		/* Notify DPM */
+		dpm_handle_phy_event(phy, PHY_DPM_EVENT_VBUS_OFF);
 	}
 }
 
diff --git a/include/linux/usb_typec_phy.h b/include/linux/usb_typec_phy.h
index a4ec743..5f107b4 100644
--- a/include/linux/usb_typec_phy.h
+++ b/include/linux/usb_typec_phy.h
@@ -116,6 +116,11 @@ enum typec_phy_evts {
 	PROT_PHY_EVENT_MSG_RCV,
 };
 
+enum typec_phy_dpm_evts {
+	PHY_DPM_EVENT_VBUS_ON,
+	PHY_DPM_EVENT_VBUS_OFF,
+};
+
 enum {
 	PD_DATA_ROLE_UFP,
 	PD_DATA_ROLE_DFP,
-- 
1.9.1

